github/pkg/adapter/adapter_test.go

```diff
@@ -407,231 +409,108 @@ var testCases = []testCase{
 	},
 }
 
-// mockTransport is a simple fake HTTP transport
-type mockTransport func(req *http.Request) (*http.Response, error)
+func newTestAdapter(t *testing.T, ce cloudevents.Client) *gitHubAdapter {
+	env := envConfig{
+		EnvConfig: adapter.EnvConfig{
+			Namespace: "default",
+		},
+		EnvSecret:    secretToken,
+		EnvPort:      "8080",
+		EnvOwnerRepo: "test.repo",
+	}
+	ctx, _ := pkgtesting.SetupFakeContext(t)
+	logger := zap.NewExample().Sugar()
+	ctx = logging.WithLogger(ctx, logger)
 
-// RoundTrip implements the required RoundTripper interface for
-// mockTransport
-func (mt mockTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	return mt(req)
+	return NewAdapter(ctx, &env, ce).(*gitHubAdapter)
 }
 
-// TestAllCases runs all the table tests
-func TestAllCases(t *testing.T) {
-	for _, tc := range testCases {
-		h := &fakeHandler{
-			//handler: tc.sink,
-			handler: sinkAccepted, // No tests expect the sink to do anything interesting
-		}
-		sinkServer := httptest.NewServer(h)
-		defer sinkServer.Close()
+func TestGracefullShutdown(t *testing.T) {
+	ce := adaptertest.NewTestClient()
+	ra := newTestAdapter(t, ce)
+	stopCh := make(chan struct{}, 1)
+
+	go func(stopCh chan struct{}) {
+		defer close(stopCh)
+		time.Sleep(time.Second)
 
-		ra, err := New(sinkServer.URL, testOwnerRepo)
+	}(stopCh)
+
+	t.Logf("starting webhook server")
+	err := ra.Start(stopCh)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestServer(t *testing.T) {
+	for _, tc := range testCases {
+		ce := adaptertest.NewTestClient()
+		adapter := newTestAdapter(t, ce)
+		hook, err := gh.New()
 		if err != nil {
-			t.Fatal(err)
+			t.Error(err)
```

Comment:
`Fatal` was correct. If this step fails, the rest of the test has no chance to succeed.

Reply:
Done.